src/containers/SignIn/SignIn.js

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { GoX } from 'react-icons/go';
+import signIn, { clearErrors } from '../../actions/SignIn';
+import styles from './SignIn.module.css';
+import SocialSignup from '../../components/SocialSignup';
+
+export class SignIn extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      email: "",
+      password: "",
+    }
+    this.signInTitle = "Welcome to Authors Haven";
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.toggleModal = this.toggleModal.bind(this);
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    const { props } = this
+    props.dispatch(signIn(this.state));
+  }
+
+  handleChange(event) {
+    this.setState({ [event.target.name]: event.target.value.trim() });
+  }
+  
+  toggleModal() {
+    const { props } = this
+    props.dispatch(clearErrors());
+    const { toggleModal } = this.props;
+    toggleModal();
+  }
+  
+  render() {
+    const { error, signingIn } = this.props;
+    return (
+      <div id={styles.regModal}>
+      <button className={styles.modalClose} type="button" onClick={this.toggleModal}>
+        <span className={styles.xIcon}>
+          <GoX />
+        </span>
+      </button>
+        <div className={styles.regContainer}>
+          <p className={styles.center} >{this.signInTitle}</p>
+
+          <SocialSignup />
+
+          {error ? <p
+          id="invalid-credential"
+          style={{ color: 'red', textAlign: 'center', marginBottom: '12px' }}>
```

Comment:
why using inline styles? you can just abstract this into the styles file

Reply:
Thanks, this is fixed now